### DISCUSSION TITLE
Accessing an entity ID from entt::connection created from on_construct (on_destroy)?

### DISCUSSION THREAD

**Discussion by hiraz01** [skypjack/entt#967] [Q&A] [answered]
Is there a way to access an entt::entity that is generating the signal when on_construct or on_destroy is called? At the moment I have something as follows:

entt::sink seqlinkOnDestroy = m_Registry.on_destroy<SeqLinkComp>()
entt::connection seqlinkConnect = seqlinkOnDestroy.connect<&SeqPanel::SeqLinkDestroyed>(m_SeqPanel)

I am trying to see if it is possible to get the entity ID of the entity whose component was destroyed when SeqLinkComp is removed and SeqLinkDestroyed() is called. I read through "Crash Course : entity-component system" discussing the reactivity of entt and I also read through "Crash Course: events, signals, and everything in between," but I am struggling to piece together the right way to think about signals in entt. In addition, I am not particularly advanced so I may be missing something obvious. 

Any help or direction would be very much appreciated.

**Comment by Green-Sky** [comment, marked as answer]
Yes, you should be able too. example code i use:
```
scene.on_destroy<Components::Position2D>().connect<&entt::registry::remove<Components::Transform4x4>>();
```
this actually calls the .remove<>(e) method on the registry.

the wiki also tells you the signature you can have:
```
void(entt::registry &, entt::entity);
```
https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#observe-changes

**Comment by hiraz01** [comment]
I do not follow your response. I have read through the link you provided, but no luck. Is it possible to retrieve the entity ID from within the connected method?

**Reply by hiraz01** [reply]
Actually, after really thinking about it and staring at your response, I tried the following:

void SeqPanel::SeqLinkDestroyed(entt::registry& entreg, entt::entity entcon){}

I was able to cast entcon to a uint32_t to display the entity ID! Thank you very much for your help.  Before I was blindly having a void() signature. By changing the signature, I could get the desired result. 

Thank you very much for your help. Much appreciated.

**Reply by Green-Sky** [reply]
I am sorry, if i formulated my answer unreadable...

> I was able to cast entcon to a uint32_t to display the entity ID

not sure if this is a permanent thing or just for debugging, but entt has a builtin cast for that `entt::to_integral(e)` :)

**Reply by hiraz01** [reply]
Thanks for the tip. Cheers.

**Reply by skypjack** [reply]
You can also use `entt::to_entity` to turn the identifier into a plain integral value. 👍

**Reply by hiraz01** [reply]
Thanks again for the help. Just out of curiosity, when you say **integral** are you using that interchangeably with integer or is this something specific to entt?

**Reply by skypjack** [reply]
Yeah, it's _integral_ as in [`is_integral`](https://en.cppreference.com/w/cpp/types/is_integral), that is, not necessarily `int` but still an integral value that you can convert to `int` if you like.
Consider that by default the underlying type for `entt::entity` is `std:.uint32_t` but you can change it if you like or just use a different entity type.
So, _integral_ is slightly more let's say general purpose and likely appropriate but it's me probably 😅 _integer_ is just fine as well. 👍
